jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -0,0 +1,2653 @@
+	}
+
+	public final void putFloatRelease(Object obj, long offset, float value) {
+		putFloatVolatile(obj, offset, value);
+	}
+
+	public final void putLongRelease(Object obj, long offset, long value) {
+		putLongVolatile(obj, offset, value);
+	}
+
+	public final void putDoubleRelease(Object obj, long offset, double value) {
+		putDoubleVolatile(obj, offset, value);
+	}
+
+	public final Object getObjectOpaque(Object obj, long offset) {
+		return getObjectVolatile(obj, offset);
+	}
+
+	public final boolean getBooleanOpaque(Object obj, long offset) {
+		return getBooleanVolatile(obj, offset);
+	}
+
+	public final byte getByteOpaque(Object obj, long offset) {
+		return getByteVolatile(obj, offset);
+	}
+
+	public final short getShortOpaque(Object obj, long offset) {
+		return getShortVolatile(obj, offset);
+	}
+
+	public final char getCharOpaque(Object obj, long offset) {
+		return getCharVolatile(obj, offset);
+	}
+
+	public final int getIntOpaque(Object obj, long offset) {
+		return getIntVolatile(obj, offset);
+	}
+
+	public final float getFloatOpaque(Object obj, long offset) {
+		return getFloatVolatile(obj, offset);
+	}
+
+	public final long getLongOpaque(Object obj, long offset) {
+		return getLongVolatile(obj, offset);
+	}
+
+	public final double getDoubleOpaque(Object obj, long offset) {
+		return getDoubleVolatile(obj, offset);
+	}
+
+	public final void putObjectOpaque(Object obj, long offset, Object value) {
+		putObjectVolatile(obj, offset, value);
+	}
+
+	public final void putBooleanOpaque(Object obj, long offset, boolean value) {
+		putBooleanVolatile(obj, offset, value);
+	}
+
+	public final void putByteOpaque(Object obj, long offset, byte value) {
+		putByteVolatile(obj, offset, value);
+	}
+
+	public final void putShortOpaque(Object obj, long offset, short value) {
+		putShortVolatile(obj, offset, value);
+	}
+
+	public final void putCharOpaque(Object obj, long offset, char value) {
+		putCharVolatile(obj, offset, value);
+	}
+
+	public final void putIntOpaque(Object obj, long offset, int value) {
+		putIntVolatile(obj, offset, value);
+	}
+
+	public final void putFloatOpaque(Object obj, long offset, float value) {
+		putFloatVolatile(obj, offset, value);
+	}
+
+	public final void putLongOpaque(Object obj, long offset, long value) {
+		putLongVolatile(obj, offset, value);
+	}
+
+	public final void putDoubleOpaque(Object obj, long offset, double value) {
+		putDoubleVolatile(obj, offset, value);
+	}
+
+	public native void unpark(Object thread);
+
+	public native void park(boolean isAbsolute, long time);
+
+	public int getLoadAverage(double[] loadavg, int numberOfElements) {
+		if ((numberOfElements < 0) || (numberOfElements > 3) || (loadavg.length > numberOfElements)) {
+			throw new ArrayIndexOutOfBoundsException();
+		}
+
+		return getLoadAverage0(loadavg, numberOfElements);
+
+	}
+
+	public final int getAndAddInt(Object obj, long index, int value) {
+		for (;;) {
+			int intAtIndex = getIntVolatile(obj, index);
```

Comment:
`weakCompareAndSetInt` should only fail if the passed in compareValue is not the same as what is written in memory which could occur since `getAndAddInt` is not atomic. If it fails we would need to try again with the new value stored in memory so updating inside the loop is necessary.